### PR TITLE
refactor icon props components for linter

### PR DESCRIPTION
**Pull request by grrowe**
- add props interface
- update `any` -> `Props`
    - optional props like `color`
- described component further with `React.FC<Props>`

### CODE REVIEW

File: src/ui/shared/icons.tsx [hunk 1]
```diff
@@ -55,8 +55,16 @@ const IconFillBase = ({
...
+  color?: string;
+  width?: string;
+  height?: string;
+  style?: object;
+  className?: string;
+}
+
 // small arrow up
-export const SmallArrowUp = (props: any) => {
+export const SmallArrowUp: React.FC<Props> = (props: Props) => {
```
Comment (neurosnap) [inline src/ui/shared/icons.tsx:67]: > In most cases, it makes very little difference which syntax is used, but the React.FC syntax is slightly more verbose without providing clear advantage, so precedence was given to the "normal function" syntax.

https://stackoverflow.com/a/59991281

It seems like we don't need `React.FC<Props>` so I would probably not add it.

File: src/ui/shared/icons.tsx [hunk 2]
```diff
@@ -55,8 +55,16 @@ const IconFillBase = ({
   );
 };
 
+export interface Props extends React.HTMLProps<HTMLDivElement> {
```
Comment (neurosnap) [inline src/ui/shared/icons.tsx:58]: I think this should be `React.SVGProps<SVGSVGElement>`: https://stackoverflow.com/a/60083905

File: src/ui/shared/icons.tsx [hunk 3]
```diff
@@ -120,7 +128,7 @@ export const ArrowsUpDown = (props: any) => {
 };
 
 // <!-- chevron-up -->
-export const ChevronUp = (props: any) => {
+export const ChevronUp: React.FC<Props> = (props: Props) => {
```
Comment (neurosnap) [inline src/ui/shared/icons.tsx:131]: There are cases where you aren't using `props`.  Please confirm all icons are using them.

**Review by neurosnap — CHANGES_REQUESTED** [review]
`IconFillBase` has `width` and `height` but it is never used in the component.

File: src/ui/shared/icons.tsx (hunk 3 above)
Comment (grrowe) [reply src/ui/shared/icons.tsx]: confirmed, all icons are using props

File: src/ui/shared/icons.tsx (hunk 2 above)
Comment (grrowe) [reply src/ui/shared/icons.tsx]: You're right, swapped it over.

File: src/ui/shared/icons.tsx (hunk 1 above)
Comment (grrowe) [reply src/ui/shared/icons.tsx]: Removed

File: src/ui/shared/icons.tsx [hunk 4]
```diff
@@ -45,8 +45,8 @@ const IconFillBase = ({
     <svg
       style={style}
       xmlns="http://www.w3.org/2000/svg"
-      width="24px"
-      height="24px"
+      width={width}
```
Comment (neurosnap) [inline src/ui/shared/icons.tsx:48]: Do we ever change the width/height?  If not let's remove the option.

Comment (grrowe) [reply src/ui/shared/icons.tsx:48]: We do not currently, but I am not sure if we will in the future or not. I will remove it now and just add it back in if we need it later.

**Review by neurosnap — APPROVED** [review]

**Pull request merged**